### PR TITLE
<optional>: Fix mystery LWG issue citations

### LINKED ISSUE #1770: `<optional>`: Mystery LWG issue citations
https://github.com/microsoft/STL/blob/dcba13b91e9f39c16319d41ab62b2b5fde2c0be5/stl/inc/optional#L417-L421
https://github.com/microsoft/STL/blob/dcba13b91e9f39c16319d41ab62b2b5fde2c0be5/stl/inc/optional#L431-L435

* Was an LWG issue filed for this?
  + If not, we should file it.
  + If so, we should cite the number.
  + If it's already been voted into the WP, we should drop "as modified by" entirely.
* We should cite a WP for `[optional.observe]`. (Paragraphs can be renumbered over time.)

### PR DESCRIPTION
**Pull request by Athira2199**
Fixes #1770.

### CODE REVIEW

**Review by miscco — APPROVED** [review]
LGTM

**Comment by miscco** [comment]
Congratulations on your first contribution 🎉

File: stl/inc/optional
Comment (CaseyCarter) [inline stl/inc/optional:421]: Pre-existing: from LWG-3424, the paragraph numbers in these citations are incorrect. They should be /17 and /19 as stated in the issue which agrees with N4885.

**Review by CaseyCarter — CHANGES_REQUESTED** [review]

File: stl/inc/optional
Comment (StephanTLavavej) [reply stl/inc/optional:421]: Oops, I didn't check the paragraph numbers, or I would have mentioned them in the issue. My mistake, so I'll push changes to fix it. :smiley_cat:

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
@Athira2199 Status update - I've pushed a commit to this PR, resolving the pre-existing issue with paragraph numbers that @CaseyCarter pointed out. (Nothing was wrong with your changes, and you perfectly followed the steps that I explained :smile_cat: - I had simply failed to notice that these paragraph numbers were off by one and also needed fixing.) By default, when you submit a PR, the maintainers of the target repo can push changes to that branch of your fork, to fix issues discovered during code review - we commonly do so when we find small issues that don't need major rework.

Our code review process needs 2 approvals for every PR, so I've asked @CaseyCarter to take another look. After he approves, we'll move your PR to the Ready To Merge column in our [Code Reviews project](https://github.com/microsoft/STL/projects/1?fullscreen=true), and then we'll merge it shortly afterwards. (Another part of our process is that we simultaneously merge changes on GitHub and the Microsoft-internal repo where MSVC is built; this is semi-manual so we batch up PRs to save work.)

**Comment by Athira2199** [comment]
@StephanTLavavej thank you for helping me out and explaining each process in detail. 😃

**Review by CaseyCarter — APPROVED** [review]
Looks good - congratulations on your first STL contribution!

**Comment by StephanTLavavej** [comment]
Thanks for fixing these citations, and congrats again! I'm recording this in the [Changelog](https://github.com/microsoft/STL/wiki/Changelog) as shipping in "Dev17" 17.0 Preview 1, but hopefully (after we "backport" commits internally) it will ship earlier in VS 2019 16.10 Preview 3 so you can see your changes in Visual Studio very soon. :smile_cat: :tada: :rocket:

**Pull request merged**
